### DISCUSSION TITLE
Overhaul of the alert and notices system

### DISCUSSION THREAD

**Discussion by pehbeh** [osTicket/roadmap#63] [Ideas]
I would like to suggest that the currently available options to be informed about new tickets or new messages from customers be revised and expanded. My idea here is that the settings can be **choose by each agent**. Currently, the settings from the department take effect. It would be great if the agent could decide this themselves. I would like to see the following notifications:

**Notify for**
- A new ticket
- A ticket is assigned to me			
- A ticket is assigned to someone else			
- I'm @mentioned in a ticket
- A ticket from my department has new updates (internal notes, due-date-updates...)
			
**Notify me when a customer replies**	 
- To an unassigned ticket			
- To a ticket assigned to me	
- To a ticket assigned by someone else

**Comment by JediKev** [comment]
@pehbeh 

I somewhat agree, however this would be on top of the administration settings. Meaning, the Admins can still control what alerts to enable globally and for each Department (maybe even each Agent). Then the Agents can toggle the available alerts. Admins still need that control of the routing and alerts but Agents should be able to further choose themselves.

But yet again it gets tricky because what if an Agent disables a critical alert on purpose, they don’t get alerted, admin wonders why the ticket never got updated, and the agent claims they never got the alert. We’ll have to think about this more in-depth because it definitely gets tricky. Maybe admins can toggle an option for each alert that determines wether or not the specific alert is configurable for agents? I don’t know.

Cheers.

**Reply by pehbeh** [reply]
> I somewhat agree, however this would be on top of the administration settings. Meaning, the Admins can still control what alerts to enable globally and for each Department (maybe even each Agent). Then the Agents can toggle the available alerts. Admins still need that control of the routing and alerts but Agents should be able to further choose themselves.

I agree with you that it can best be seen as an extension. The admin sets the basic notifications, maybe also as you said with the option to set notifications so that a agent can't disable them. With the extensions, that the agents can make further customizations for themselves on top of that.

I am also basically concerned that one setting can be too _rigid_ for all departments in the system. If a agent is in several departments but wants to follow a certain workflow it would be great if one could make additional settings as an agent.

For example: many of our employees are missing the possibility to be informed about new messages or internal notes from tickets assigned to them and other agents, because they want to keep a general overview. But such a setting can be too much for others again. This is a great way to make the existing system even more powerful and to give agents options to integrate their own workflow.

Thanks for your thoughts. :)

**Comment by gamepass234** [comment]
It would be really great if the notifications were more flexible and more available. I think the suggestion is great!
